tests/osquery/test_api_views.py

```diff
@@ -1091,3 +1120,257 @@ def test_export_distributed_query_results_ok(self):
+        self.assertEqual(query.name, new_name)
+
+    def test_update_query_unauthorized(self):
+        query = self.force_query()
+        new_name = get_random_string(12)
+        data = {"name": new_name, "sql": query.sql}
+        response = self.put_json_data(reverse("osquery_api:query", args=(query.pk,)), data, include_token=False)
+        self.assertEqual(response.status_code, 401)
+
+    def test_update_query_permission_denied(self):
+        query = self.force_query()
+        new_name = get_random_string(12)
+        data = {"name": new_name, "sql": query.sql}
+        response = self.put_json_data(reverse("osquery_api:query", args=(query.pk,)), data)
+        self.assertEqual(response.status_code, 403)
+
+    def test_update_query_ztl_status_validate_error(self):
+        query = self.force_query()
+        data = {"name": query.name, "sql": query.sql, "compliance_check_enabled": True}
+        self.set_permissions("osquery.change_query")
+        response = self.put_json_data(reverse("osquery_api:query", args=(query.pk,)), data)
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json(), {'compliance_check_enabled': ['ztl_status not in sql']})
+
+    def test_update_query_ztl_status_validate_success(self):
+        query = self.force_query()
+        data = {"name": query.name, "sql": "ztl_status;", "compliance_check_enabled": True}
+        self.set_permissions("osquery.change_query")
+        response = self.put_json_data(reverse("osquery_api:query", args=(query.pk,)), data)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json()["compliance_check_enabled"], True)
+
+    def test_update_query_increment_version(self):
+        query = self.force_query()
+        self.assertEqual(query.version, 1)
+        new_sql = "changed sql line;"
+        data = {"name": query.name, "sql": new_sql}
+        self.set_permissions("osquery.change_query")
+        response = self.put_json_data(reverse("osquery_api:query", args=(query.pk,)), data)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json()["version"], 2)
+        query.refresh_from_db()
+        self.assertEqual(query.version, 2)
+
+    def test_update_query_with_pack_query_snapshot_mode_validate_success(self):
+        query = self.force_query(pack_query_mode="snapshot", compliance_check=False)
+        pack_query = PackQuery.objects.get(slug=slugify(query.name))
+        data = {"name": query.name, "sql": "select 'OK' as ztl_status;", "compliance_check_enabled": True}
+        self.set_permissions("osquery.change_query")
+        response = self.put_json_data(reverse("osquery_api:query", args=(query.pk,)), data)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json()["compliance_check_enabled"], True)
+        self.assertEqual(pack_query.slug, slugify(query.name))
+        self.assertIs(pack_query.snapshot_mode, True)
```

Comment:
No need to test pack_query.snapshot_mode, since it is in the helper method, not what the API is doing.